pico-slack.js

```diff
@@ -66,6 +66,25 @@ const processEvent = (event)=>{
 	return evt;
 };
 
+const handleSocketClose = ()=>{
+	if (Slack.closing) return; // Socket was closed intentionally.
+
+	Slack.connected = false;
+	if (!Slack.autoReconnect) {
+		Slack.error('WebSocket closed unexpectedly!');
+		return;
+	}
+
+	Slack.warn('WebSocket closed unexpectedly, trying to reconnect...');
+	return Slack.connect(Slack.token)
+		.then(()=>Slack.log('Reconnected!'))
+		.catch((err)=>Slack.error(`Failed to reconnect: ${err}`, err));
+};
+
+const handleSocketError = (err)=>{
+	Slack.error(`Slack socket error: ${err}`, err);
```

Comment:
I'm not sure if the library should call `log`/`warn`/`error` on its own, as this could cause unwanted output to be sent to the user's Slack. Obviously, we do want this to happen in our case, but maybe that should be handled within the Higgins repo. Or maybe there should be a `debug`/`verbose` flag that enables this output?

Reply:
That's a great point.  I'll pull these out into the subsequent Higgins PR.

At first I was worried I wouldn't be able to since I wanted progress reports in Slack (trying to reconnect -> reconnected/failed), and I wouldn't want to expose those progress steps outside pico-slack.  But now that we know _why_ the disconnects are happening, I don't really care to have the progress reports anymore, so I'm just gonna drop those ones.